### PR TITLE
Feature/10464 crypto functions part 2/10684 integration

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Integrates all the parts in the ticket validation class and fixes some issues in used components.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10684

### CODE REVIEW

**Review by CV113 — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Scenes/TicketValidation/Result/__tests__/TicketValidationResultViewModelTests.swift [hunk 1]
```diff
@@ -9,6 +9,7 @@ class TicketValidationResultViewModelTests: XCTestCase {
 
 	func testGIVEN_ValidationResults_WHEN_PassedModelIsCreated_THEN_ModelIsSetupCorrectly() throws {
 		let model = TicketValidationPassedViewModel(
+			validationDate: Date(),
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Scenes/TicketValidation/Result/__tests__/TicketValidationResultViewModelTests.swift:12]: 👍

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift [hunk 2]
```diff
@@ -24,61 +24,167 @@ final class TicketValidation: TicketValidating {
...
+					completion(.failure(.resultToken(error)))
+				}
 			}
+		case .failure(let error):
+			completion(.failure(.encryption(error)))
 		}
 	}
 
+	func cancel() {
+
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift:172]: We don't call server in this case?

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/TicketValidation/TicketValidation.swift:172]: The story for graceful cancellation was postponed, so the implementation is empty for now

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift [hunk 3]
```diff
@@ -270,7 +270,6 @@ enum AccessibilityIdentifiers {
 			static let dataPrivacy = "QRScanner_Info_dataPrivacy"
 		}
 
-		#if targetEnvironment(simulator)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:273]: Just curious why?

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/ECKeyPairGeneration.swift [hunk 4]
```diff
@@ -11,11 +12,17 @@ public enum ECKeyPairGenerationError: Error {
 }
 
 public struct ECKeyPairGeneration {
+
+    // MARK: - Init
+
+    public init() {
```
Comment (marcussc) [inline src/xcode/ENA/ENASecurity/Sources/ENASecurity/ECKeyPairGeneration.swift:18]: Can't this be removed? 🤔

File: src/xcode/ENA/ENA/Source/Services/TicketValidation/MockTicketValidation.swift [hunk 5]
```diff
@@ -4,6 +4,8 @@
 
 import Foundation
 
+#if DEBUG
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Services/TicketValidation/MockTicketValidation.swift:7]: @kaddaSz FYI: This should fix the xmake issue.

File: src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/View Helpers/AccessibilityIdentifiers.swift:273]: With this tests can't be run on a device anymore, and we had a unit test that we wonted to run on a device as it actually produced different results there

File: src/xcode/ENA/ENASecurity/Sources/ENASecurity/ECKeyPairGeneration.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENASecurity/Sources/ENASecurity/ECKeyPairGeneration.swift:18]: Need this to make it public

**Review by marcussc — APPROVED** [review]

**Pull request merged**